models/P2PPresenter/effects.js

```diff
@@ -0,0 +1,73 @@
+const ps = require('peer-star')
+
+module.exports = globalConfig => ({
+  createStar: createStar(globalConfig),
+  broadcast: broadcast,
+  sendTo: sendTo
+})
+
+function createStar (globalConfig) {
+  return inner
+  function inner (state, group, send, done) {
+    var opts = {
+      hubURL: globalConfig.hub,
+      GID: group,
+      isMain: true
+    }
+    state.star = ps(opts)
+
+    state.star.on('peer', (peer, id) => {
+      console.log('connected to a new peer:', id)
+      console.log('total peers:', state.star.peers.length)
+      send('presenter:clientAdded', {peer: peer, id: id}, (err, res) => { if (err) done(err) })
+      addListenersToPeer(send, peer, id)
+    })
+    state.star.on('disconnect', (peer, id) => {
+      console.log('disconnected from a peer:', id)
+      console.log('total peers:', state.star.peers.length)
+      peer.destroy()
+      send('presenter:clientLeft', {peer: peer, id: id}, (err, res) => { if (err) done(err) })
+    })
+    done()
+  }
+}
+
+function addListenersToPeer (send, peer, id) {
+  peer.on('data', function (data) {
+    var decoded
```

Comment:
I didn't use let because I am not familiar with all the effects that come with declaring vars with `let`.
I agree that `decoded` could be const. The scope for the variable is very short, so it should be fine as var.